lib/spack/spack/cmd/__init__.py

```diff
@@ -109,15 +106,19 @@ def parse_specs(args, **kwargs):
             elif normalize:
                 spec.normalize()
 
-        return specs
-
     except spack.parse.ParseError as e:
-        tty.error(e.message, e.string, e.pos * " " + "^")
-        sys.exit(1)
+        tty.die(e.message, e.string, e.pos * " " + "^")
 
     except spack.spec.SpecError as e:
-        tty.error(e.message)
-        sys.exit(1)
+        tty.die(e.message)
+
+    if allow_multi:
+        return specs
+
+    if len(specs) != 1:
+        tty.die('only one spec can be installed at a time.')
+    spec = specs.pop()
```

Comment:
I don't get the change in arguments: why `allow-multi`? `parse specs` is supposed to be able to parse multiple specs. The fact that currently it is not is a bug (see #1248 and #1976).

Also, an error message that says that only one item can be *installed* doesn't seem appropriate in a function that *parses*.